smartseq2_pipeline/run_multi_samples/scripts/parse_picard.py

```diff
@@ -0,0 +1,77 @@
+from crimson import picard
+import json
+import pandas as pd
+import argparse
+import sys
+from collections import OrderedDict
+import os
+
+def parse_picard_to_table_by_sample(input_metrics_list,output_filename):
+    mets=OrderedDict()
+    with open(input_metrics_list) as f:
```

Comment:
Can this code that reads the `inputs_metrics_list` file get pulled out into a helper function? It's also used in `parse_picard_to_table_by_type` so that can removed the duplicated code.

Reply:
I thought about it before. the problem is OrderDict can not handle nest ordered dict well.  for the merging by sample, the order of keys is critical but for merging by metrics, it does not matter. That is why I separate them

Reply:
Oh okay! Do you think some of the code that parses the file to get `met` could be separated out? The less duplication the better :grin:

Reply:
I probably will use them somewhere  when I do benchmarking. that is why i want to put together in one script.

Reply:
Oh ok!